### PR TITLE
doc: west fixes and updates to manifest.rst

### PR DESCRIPTION
**Pull request by mbolivar**
Some of this information is stale and needs to be fixed.

Try to make a few other things clearer while we're here.

### CODE REVIEW

File: doc/guides/west/manifest.rst [hunk 1]
```diff
@@ -156,82 +115,105 @@ bases are respectively ``https://example.com/base1`` and
...
-Each element in the ``projects`` sequence can contain the following keys. Some
-of the description refers to the ``defaults`` subsection, which will be
-described next.
-
-- ``name``: Mandatory, the name of the project. The fetch URL is formed as
-  remote url-base + '/' + ``name``. The name cannot be one of the reserved
-  values "west" and "manifest".
-- ``remote``: The name of the project's remote. If not given, the ``remote``
-  value in the ``defaults`` subsection is tried next. If both are missing, the
+  ``https://example.com/base1/proj1``. The remote ``url-base`` plus a ``/`` is
```
Comment (dbkinder) [inline doc/guides/west/manifest.rst]: Clearer would be:
```
The remote ``url-base`` is  appended with a ``/`` and
the project ``name`` to form the URL.
```

File: doc/guides/west/manifest.rst [hunk 2]
```diff
@@ -156,82 +115,105 @@ bases are respectively ``https://example.com/base1`` and
...
+  be checked out when it is next updated.
+
+The list of project keys and their usage follows. Sometimes we'll refer to the
+``defaults`` subsection; it will be described next.
+
+- ``name``: Mandatory, the name of the project. The name cannot be one of the
+  reserved values "west" or "manifest". The name must be unique in the manifest
+  file.
+- ``remote`` or ``url``:
+
```
Comment (dbkinder) [inline doc/guides/west/manifest.rst]: you should delete this blank line to be consistent with the other list items.

File: doc/guides/west/manifest.rst [hunk 3]
```diff
@@ -156,82 +115,105 @@ bases are respectively ``https://example.com/base1`` and
...
+- ``proj3`` has an explicit ``url``, so it will be fetched from
+  ``https://github.com/user/project-three``.
+
+  Its local path defaults to its name, ``proj3``. Commit ``abcde413a111`` will
+  be checked out when it is next updated.
+
+The list of project keys and their usage follows. Sometimes we'll refer to the
+``defaults`` subsection; it will be described next.
+
+- ``name``: Mandatory, the name of the project. The name cannot be one of the
```
Comment (dbkinder) [inline doc/guides/west/manifest.rst]: You use the word ``Optional.`` below, so you should be consistent here by writing ``Mandatory.  The name of the project.``

**Review by dbkinder — CHANGES_REQUESTED** [review]
Looks better.  A few suggestions.

File: doc/guides/west/manifest.rst (hunk 2 above)
Comment (mbolivar) [reply doc/guides/west/manifest.rst]: I actually inserted it on purpose to emphasize that it's not like the others: it's the only one which describes two different keys.

I can make that change if you still think it's a good idea.

**Comment by mbolivar** [comment]
@dbkinder thanks for the review! I've responded to your comments.

File: doc/guides/west/manifest.rst (hunk 2 above)
Comment (mbolivar) [reply doc/guides/west/manifest.rst]: Looking at this again, I've replaced with:

```
- ``remote`` or ``url``: Mandatory (one of the two, but not both).
```

**Review by dbkinder — APPROVED** [review]
LGTM, thanks!

**Pull request merged**